app/services/theme_settings_migrations_runner.rb

```
@@ -0,0 +1,174 @@
+        name: migration.original_name,
+        error: error.message,
+      )
+    rescue MiniRacer::V8OutOfMemoryError
+      raise_error(
+        "themes.import_error.migrations.exceeded_memory_limit",
+        name: migration.original_name,
+      )
+    rescue MiniRacer::ScriptTerminatedError
+      raise_error("themes.import_error.migrations.timed_out", name: migration.original_name)
+    rescue MiniRacer::RuntimeError => error
+      message = error.message
+      if message.include?("no_exported_migration_function")
+        raise_error(
+          "themes.import_error.migrations.no_exported_function",
+          name: migration.original_name,
+        )
+      elsif message.include?("default_export_is_not_a_function")
+        raise_error(
+          "themes.import_error.migrations.default_export_not_a_function",
+          name: migration.original_name,
+        )
+      elsif message.include?("migration_function_no_returned_value")
+        raise_error(
+          "themes.import_error.migrations.no_returned_value",
+          name: migration.original_name,
+        )
+      elsif message.include?("migration_function_wrong_return_type")
+        raise_error(
+          "themes.import_error.migrations.wrong_return_type",
+          name: migration.original_name,
+        )
+      else
+        raise_error(
+          "themes.import_error.migrations.runtime_error",
+          name: migration.original_name,
+          error: message,
+        )
+      end
+    end
+  end
+
+  private
+
+  def lookup_pending_migrations_fields
+    @theme
+      .migration_fields
+      .left_joins(:theme_settings_migration)
+      .where(theme_settings_migration: { id: nil })
+  end
+
+  def convert_fields_to_migrations(fields)
+    fields.map do |field|
+      match_data = /\A(?<version>\d{4})-(?<name>.+)/.match(field.name)
+
+      if !match_data
+        raise_error("themes.import_error.migrations.invalid_filename", filename: field.name)
+      end
```

I don't think we should be doing the check here since validation is already done when importing the theme.

---

I feel like it doesn't hurt to keep the check here. We need to do the regex match to extract the number and name, and if somehow the regex doesn't match, I think it's better to raise an error with clear/helpful message than to raise a less-helpful 500 error.